public/app/plugins/datasource/testdata/components/SimulationSchemaForm.tsx

```diff
@@ -0,0 +1,77 @@
+import { css } from '@emotion/css';
+import React, { FormEvent, useState } from 'react';
+
+import { DataFrameSchema, FieldSchema, GrafanaTheme2 } from '@grafana/data';
+import { useStyles2, TextArea, InlineField, Input, FieldSet, InlineSwitch } from '@grafana/ui';
+
+interface SchemaFormProps {
+  config: Record<string, any>;
+  schema: DataFrameSchema;
+  onChange: (config: Record<string, any>) => void;
+}
+
+const renderInput = (field: FieldSchema, onChange: SchemaFormProps['onChange'], config: SchemaFormProps['config']) => {
+  switch (field.type) {
+    case 'number':
+      return (
+        <Input
+          type="number"
+          defaultValue={config?.[field.name]}
+          onChange={(e: FormEvent<HTMLInputElement>) => {
+            const newValue = e.currentTarget.valueAsNumber;
+            onChange({ ...config, [field.name]: newValue });
+          }}
+        />
+      );
+    case 'boolean':
+      return (
+        <InlineSwitch
+          value={config?.[field.name] ?? true}
+          onChange={() => {
+            onChange({ ...config, [field.name]: !config[field.name] });
+          }}
+        />
+      );
+    default:
+      return <></>;
+  }
+};
+export const SimulationSchemaForm = ({ config, schema, onChange }: SchemaFormProps) => {
+  const [jsonView, setJsonView] = useState<boolean>(false);
+
+  const getStyles = (theme: GrafanaTheme2) => {
```

Comment:
I have seen the `getStyles` function outside the component scope before... i think that pattern avoids recreating it on each render :)